### PR TITLE
[RMS]: advanced queries data sources

### PR DESCRIPTION
**Pull request by artem-lifshits**
## Summary of the Pull Request
New data source ``data_source/opentelekomcloud_rms_advanced_query_v1`` 
New data source ``data_source/opentelekomcloud_rms_advanced_query_schemas_v1``

## PR Checklist

* [x] Refers to: #2659
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
2024/11/06 14:48:04 [INFO] Building Swift S3 auth structure
2024/11/06 14:48:04 [INFO] Setting AWS metadata API timeout to 100ms
2024/11/06 14:48:06 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2024/11/06 14:48:06 [INFO] Swift S3 Auth provider used: "StaticProvider"
=== RUN   TestAccDataSourceAdvancedQuery_basic
=== PAUSE TestAccDataSourceAdvancedQuery_basic
=== CONT  TestAccDataSourceAdvancedQuery_basic
--- PASS: TestAccDataSourceAdvancedQuery_basic (95.56s)
PASS

Process finished with the exit code 0

2024/11/06 15:08:20 [INFO] Building Swift S3 auth structure
2024/11/06 15:08:20 [INFO] Setting AWS metadata API timeout to 100ms
2024/11/06 15:08:21 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2024/11/06 15:08:21 [INFO] Swift S3 Auth provider used: "StaticProvider"
=== RUN   TestAccDataSourceRmsAdvancedQuerySchemas_basic
=== PAUSE TestAccDataSourceRmsAdvancedQuerySchemas_basic
=== CONT  TestAccDataSourceRmsAdvancedQuerySchemas_basic
--- PASS: TestAccDataSourceRmsAdvancedQuerySchemas_basic (95.58s)
PASS

Process finished with the exit code 0
```

### CODE REVIEW

File: opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	mErr := &multierror.Error{}
+
+	// Устанавливаем отфильтрованные схемы
+	if err := d.Set("schemas", flattenSchemas(filteredSchemas)); err != nil {
+		mErr = multierror.Append(mErr, fmt.Errorf("error setting schemas: %s", err))
+	}
+
+	// Генерируем UUID для ресурса
```
Comment (muneeb-jan) [inline opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go]: comments?

File: opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	res, err := advanced.ListSchemas(client, advanced.ListSchemasOpts{
+		DomainId: GetRmsDomainId(client, config),
+	})
+	if err != nil {
+		return diag.Errorf("error getting the advanced query schemas list from server: %s", err)
+	}
+
+	schemaType := d.Get("type").(string)
+
+	// Фильтруем схемы по типу, если тип указан
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go]: wow cyrillic, better to omit

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go (hunk 1 above)
Comment (artem-lifshits) [reply opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go]: fixed

File: opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go (hunk 2 above)
Comment (artem-lifshits) [reply opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go]: yeah I prefer chinese comments but cyrillic is nice for variety..

**Review by anton-sidelnikov — APPROVED** [review]

**Review by muneeb-jan — APPROVED** [review]

**Review by Aloento — APPROVED** [review]

**Pull request merged**